### DISCUSSION TITLE
tutorials/3d/procedural_geometry/immediatemesh

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#102] [User-contributed notes]
# tutorials/3d/procedural_geometry/immediatemesh

The ImmediateMesh is a convenient tool to create dynamic geometry using an OpenGL 1.x-style API. Which makes it both approachable to use and efficient for meshes which need to be updated every fram...

https://docs.godotengine.org/en/stable/tutorials/3d/procedural_geometry/immediatemesh.html

**Comment by RonYanDaik** [comment]
How to use same material for every `surface_begin` when using mesh.surface_begin(Mesh.PRIMITIVE_TRIANGLE_STRIP,mat)?
